### ISSUE TITLE
SKGLView does not dispose or release memory [Xamarin Forms UWP]

### ISSUE THREAD

**Issue by comfr3ak** [mono/SkiaSharp#705]
### Description

When using SKGlView in UWP on a page in a grid and then navigating away from that page, the SKGLView never disposes. If you create a new page with a new SKGLView they just keep building up in memory. There is no dispose method on SKGLView to release the underlying UWP native drawing objects.

### Expected Behavior

After the page is gone from the navigation stack and the SKGLView is disconnected from the PaintSurface event, it should release and dispose with the page. 

### Actual Behavior

SKGLView remains in memory and each time a new one is created, none of them dispose themselves according to VS2017 memory profiler. 

### Basic Information

- Version with issue: 1.60.3
- Last known good version:  
- IDE:  VS2017 
- Platform Target Frameworks: UWP
  - Android:  
  - iOS:  
  - Linux:  
  - macOS:  
  - Tizen:   
  - tvOS:  
  - UWP:   
  - watchOS:  
  - Windows Classic:  
- Target Devices:
  -   

### Reproduction Link

> VS bug [#735693](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/735693)

**Comment by charlesroddie** [comment]
We are also seeing this. CC @sir-deenicus

**Comment by charlesroddie** [comment]
This also applies to 1.68.0.

This makes SkiaSharp an extremely large memory leak.

@mattleibow do you have any ideas?

This may block our release so if we can help please tell us where to get started.

It's a little surprising that SKGLView is not IDisposable, since a lot of other SkiaSharp objects are.

**Comment by mattleibow** [comment]
Thanks for reporting this, I am having a look at this. I'll let you know what I find, and if there are any workarounds/solutions.

**Comment by charlesroddie** [comment]
There is some work exploring this in the CSharpMath repo: https://github.com/verybadcat/CSharpMath/issues/23

**Comment by mattleibow** [comment]
Thanks for that issue. I'll see if I can get back to you today.

**Comment by mattleibow** [comment]
I am having a look at this, and the sample attached to the CSharpMath issue has cycles - the renderer keeps a reference to the native view which has a reference to the renderer, and the cycle is complete.

But, this is v1.60.0, which had that issue. I am trying to see what happens with v1.60.3 and also with v1.68.0.

**Comment by mattleibow** [comment]
Although, something is happening here...

I know this is the SKCanvasView... Will try the SKGLView in a sec.

The first block is the start/baseline. Then I add a SKCanvasView, and it goes up high. I cleared the view and did a snapshot - for some reason it still had 40 items. Then, 3 seconds later, it magically cleared up again:

![image](https://user-images.githubusercontent.com/1096616/50356492-c877ca00-055a-11e9-83c5-3050fa564238.png)

**Comment by mattleibow** [comment]
Yay! 🎉🎉🎉 Found the issue - I had to dig deep! Not really, just a lot of testing.
What was happening was the view was initializing ANGLE, and then releasing all the references and things - except for the final context. This left all sorts of references on the native side still hooked up _inside_ the ANGLE library. As a result, even though the managed GC was all free, ANGLE was never told that it should let go of the views.

**Comment by mattleibow** [comment]
The new graph! I took the same sample, and upgraded to my "preview" NuGets and this is what we see:

![image](https://user-images.githubusercontent.com/1096616/50438480-d335b480-08f6-11e9-962d-300b86120891.png)

**Comment by comfr3ak** [comment]
How can I get the fix within vs2017? Is there a new preview version to
install from nuget?

**Comment by mattleibow** [comment]
I am working on merging a few branches and then I hope to get a preview release out. Can't say for sure, but might be next week sometime.

**Comment by mattleibow** [comment]
@comfr3ak @charlesroddie @Happypig375 I have pushed out a preview release to our Early Access Program feed - let me know if this works for you: https://www.myget.org/F/skiasharp-eap/api/v3/index.json

**Comment by Happypig375** [comment]
There is a new crash with 1.68.1-preview71.

Reproduction:

1. Start debugging skia-leak-min (the sample in CSharpMath)
2. Click "GL View Mathview"
3. Wait for UI to respond
4. Click "GL View Mathview" again
5. Wait for debugger to break
6. Debugger breaks at 
```cs
#if DEBUG && !DISABLE_XAML_GENERATED_BREAK_ON_UNHANDLED_EXCEPTION
UnhandledException += (sender, e) =>
{
    if (global::System.Diagnostics.Debugger.IsAttached) global::System.Diagnostics.Debugger.Break();
};
#endif
```

<summary>
Exception message:
</summary>

```
System.Exception: Failed to create EGL surface
   at SkiaSharp.Views.GlesInterop.GlesContext.CreateSurface(SwapChainPanel panel, Nullable`1 renderSurfaceSize, Nullable`1 resolutionScale)
   at SkiaSharp.Views.UWP.AngleSwapChainPanel.EnsureRenderSurface()
   at SkiaSharp.Views.UWP.AngleSwapChainPanel.OnSizeChanged(Object sender, SizeChangedEventArgs e)
```

**Comment by mattleibow** [comment]
Darn, let me have a look.

**Comment by mattleibow** [comment]
This was fixed in #1024
